### PR TITLE
Destination Redshift: Add generation_id and sync_id

### PR DESCRIPTION
**Pull request by edgao**
* upgrade cdk to the next breakpoint (includes gen_id/sync_id, and adds plumbing for stream statuses, but doesn't yet include any actual refreshes logic)
     * uses a backported CDK version - https://github.com/airbytehq/airbyte/pull/40499
* also: includes the better defaultNamespace handling / stream status plumbing from the CDK bump

ran two syncs in the perf test workspace with this version; verified that the tables got generation_id correctly

closes https://github.com/airbytehq/airbyte-internal-issues/issues/8358

### CODE REVIEW

**Comment by edgao** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/40201?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#40201** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/40201?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#39008** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/39008?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#39334](https://github.com/airbytehq/airbyte/pull/39334) <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/39334?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#39141** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/39141?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#38893** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38893?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#38886** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38886?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @edgao and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-integrations/connectors/destination-redshift/src/main/kotlin/io/airbyte/integrations/destination/redshift/typing_deduping/RedshiftDestinationHandler.kt [hunk 1]
```diff
@@ -90,12 +90,9 @@ class RedshiftDestinationHandler(
         for (transaction in transactions) {
             val transactionId = UUID.randomUUID()
             if (logStatements) {
-                log.info(
-                    "Executing sql {}-{}: {}",
-                    queryId,
-                    transactionId,
-                    java.lang.String.join("\n", transaction)
-                )
+                log.info {
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-redshift/src/main/kotlin/io/airbyte/integrations/destination/redshift/typing_deduping/RedshiftDestinationHandler.kt:93]: these log statements generated a compiler warning, so switch to the non-deprecated version

File: airbyte-integrations/connectors/destination-redshift/src/main/kotlin/io/airbyte/integrations/destination/redshift/typing_deduping/RedshiftDestinationHandler.kt [hunk 2]
```diff
@@ -90,12 +90,9 @@ class RedshiftDestinationHandler(
...
+                    "Executing sql $queryId-$transactionId: ${java.lang.String.join("\n", transaction)}"
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-redshift/src/main/kotlin/io/airbyte/integrations/destination/redshift/typing_deduping/RedshiftDestinationHandler.kt:94]: use transaction.joinToString("\n") instead

File: airbyte-integrations/connectors/destination-redshift/src/main/kotlin/io/airbyte/integrations/destination/redshift/typing_deduping/RedshiftDestinationHandler.kt [hunk 3]
```diff
@@ -113,7 +110,7 @@ class RedshiftDestinationHandler(
                     logStatements = logStatements
                 )
             } catch (e: SQLException) {
-                log.error("Sql {}-{} failed", queryId, transactionId, e)
+                log.error { "Sql $queryId-$transactionId failed" }
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-redshift/src/main/kotlin/io/airbyte/integrations/destination/redshift/typing_deduping/RedshiftDestinationHandler.kt:113]: log.error(e) {...}

**Review by stephane-airbyte — APPROVED** [review]

**Pull request merged**